js/portfolio/portfolioView.js

```diff
@@ -0,0 +1,157 @@
+// Configure a view object, to hold all our functions for dynamic updates and article-related event handlers.
+var portfolioView = {};
+
+portfolioView.populateFilters = function() {
+  $('article').each(function() {
+    if (!$(this).hasClass('template')) {
+
+      var val = $(this).find('.section-date').data('date');
+      val = new Date(val);
+      val = val.getFullYear();
+      var optionTag = '<option value="' + val + '">' + val + '</option>';
+      if ($('#date-filter option[value="' + val + '"]').length === 0) {
+        $('#date-filter').append(optionTag);
+      }
+
+      val = $(this).attr('data-category');
+      optionTag = '<option value="' + val + '">' + val + '</option>';
+      if ($('#category-filter option[value="' + val + '"]').length === 0) {
+        $('#category-filter').append(optionTag);
+      }
+    }
+  });
+};
+
+portfolioView.handleDateFilter = function() {
+  $('#date-filter').on('change', function() {
+    if ($(this).val()) {
+      // TODO: If the select box was changed to an option that has a value, we need to:
```

Comment:
Don't just copy code from the pair assignment. It's better for your learning if you retype it out, even if you're keeping it the same. Getting the muscle memory down is also important.